test/gtest/ucp/test_ucp_stream.cc

```diff
@@ -33,12 +33,13 @@ class test_ucp_stream_base : public ucp_test {
 
 size_t test_ucp_stream_base::wait_stream_recv(void *request)
 {
+    ucs_time_t deadline = ucs::get_deadline();
     ucs_status_t status;
     size_t       length;
     do {
         progress();
         status = ucp_stream_recv_request_test(request, &length);
-    } while (status == UCS_INPROGRESS);
+    } while ((status == UCS_INPROGRESS) && (ucs_get_time() < deadline));
```

Comment:
why is it needed? doesn't it fail on next line assert anyway?

Reply:
assert below checks that request is completed successfully, deadline prevents possible hang in case if request is not completed at all.